src/memtable.rs

```diff
@@ -36,9 +36,9 @@ pub struct EntryIndex {
     pub compression_type: CompressionType,
 
     /// The relative offset within its group of entries.
-    pub entry_offset: u64,
+    pub entry_offset: u32,
```

Comment:
wrap `ThinEntryIndex` directly?

Reply:
No because `ThinEntryIndex` should be a private struct internal to `MemTable`.